src/backend/columnar/columnar_customscan.c

```diff
@@ -228,6 +244,177 @@ IsNotIndexPath(Path *path)
 }
 
 
+/*
+ * RecostColumnarIndexPaths re-costs index paths of given RelOptInfo for
+ * columnar table with relationId.
+ */
+static void
+RecostColumnarIndexPaths(PlannerInfo *root, RelOptInfo *rel, Oid relationId)
+{
+	Path *path = NULL;
+	foreach_ptr(path, rel->pathlist)
+	{
+		if (IsA(path, IndexPath))
```

Comment:
Should we include a comment about bitmap index scans here?

Reply:
done